actors/builtin/miner/miner_actor.go

```diff
@@ -253,34 +255,8 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
 		provenSectors, err := abi.BitFieldUnion(partitionsSectors...)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to union %d partitions of sectors", len(partitionsSectors))
 
-		// TODO WPOST (follow-up): process Skipped as faults
-
-		// Extract a fault set relevant to the sectors being submitted, for expansion into a map.
-		declaredFaults, err := bitfield.IntersectBitField(provenSectors, st.Faults)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to intersect proof sectors with faults")
-
-		declaredRecoveries, err := bitfield.IntersectBitField(declaredFaults, st.Recoveries)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to intersect recoveries with faults")
-
-		expectedFaults, err := bitfield.SubtractBitField(declaredFaults, declaredRecoveries)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to subtract recoveries from faults")
-
-		nonFaults, err := bitfield.SubtractBitField(provenSectors, expectedFaults)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to diff bitfields")
-
-		empty, err := nonFaults.IsEmpty()
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to check if bitfield was empty")
-		if empty {
-			rt.Abortf(exitcode.ErrIllegalArgument, "no non-faulty sectors in partitions %s", params.Partitions)
-		}
-
-		// Select a non-faulty sector as a substitute for faulty ones.
-		goodSectorNo, err := nonFaults.First()
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to get first good sector")
-
-		// Load sector infos for proof
-		sectorInfos, err := st.LoadSectorInfosWithFaultMask(store, provenSectors, expectedFaults, abi.SectorNumber(goodSectorNo))
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load sector infos")
+		sectorInfos, declaredRecoveries, err := st.LoadSectorInfosForProof(store, provenSectors)
```

Comment:
This is a straight-up move of logic out of the middle of this  method to a place where things that are submitting window posts can reach it. @acruikshank and @magik6k should be able to delete some copy-pasta from the window post scheduling and just call this method, now.